gallery/gallery/lib/pages/demo.dart

```diff
@@ -314,46 +316,15 @@ class _DemoPageState extends State<DemoPage> with TickerProviderStateMixin {
       buildRoute: _currentConfig.buildRoute,
     );
     if (isDesktop) {
-      // If the available width is not very wide, reduce the amount of space
-      // between the demo content and the selected section.
-      const reducedMiddleSpaceWidth = 48.0;
-
-      // Width of the space between the section and the demo content
-      // when the code is NOT displayed.
-      final nonCodePageMiddleSpaceWidth = mediaQuery.size.width > 900
-          ? horizontalPadding
-          : reducedMiddleSpaceWidth;
-
-      // Width of the space between the section and the demo content
-      // when the code is displayed.
-      final codePageMiddleSpaceWidth =
-          min(reducedMiddleSpaceWidth, nonCodePageMiddleSpaceWidth);
-
-      // Width of the space between the section and the demo content
-      final middleSpaceWidth = _state == _DemoState.code
-          ? codePageMiddleSpaceWidth
-          : nonCodePageMiddleSpaceWidth;
-
-      // Width for demo content.
-      // It is calculated in this way because the code demands more space.
-      final demoContentWidth = (mediaQuery.size.width -
-              horizontalPadding * 2 -
-              nonCodePageMiddleSpaceWidth) /
-          2;
-
-      final Widget sectionAndDemo = (_state == _DemoState.fullscreen)
-          ? demoContent
-          : Row(
-              crossAxisAlignment: CrossAxisAlignment.start,
-              children: [
-                Expanded(child: section),
-                SizedBox(width: middleSpaceWidth),
-                Container(
-                  width: demoContentWidth,
-                  child: demoContent,
-                ),
-              ],
-            );
+      final isFullScreen = _state == _DemoState.fullscreen;
+      final Widget sectionAndDemo = Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          if (!isFullScreen) Expanded(child: section),
+          SizedBox(width: !isFullScreen ? 48.0 : 0),
```

Comment:
I think this might be easier to read as:

```
if (!isFullScreen) ...[
  Expanded(child: section),
  SizedBox(),
]
Expanded(child: demoContent),
```

Reply:
This is what I was saying causes the demoContent to be rebuilt from scratch. I could write it as 

```dart
if (!isFullScreen) ...[
  Expanded(child: section),
  SizedBox(),
]
Expanded(child: demoContent),
```
and it works

Reply:
Very strange...ok I think I prefer what you originally had instead of the 2nd sizedbox